components/automate-ui/src/app/modules/user/user-table/user-table.component.spec.ts

```diff
@@ -41,12 +31,26 @@ describe('UserTableComponent', () => {
     expect(component).toBeTruthy();
   });
 
-  describe('when a baseUrl is passed', () => {
-    it('should populate the permission paths', () => {
+  describe('when {get,create}PermissionsPath are not passed', () => {
+    it('should populate the permission paths using baseUrl', () => {
+      component.baseUrl = baseUrl;
+      component.ngOnInit();
       fixture.detectChanges();
-      expect(component.getPermissionsPath).toEqual([baseUrl, 'get']);
-      expect(component.createPermissionsPath).toEqual([baseUrl, 'post']);
+      expect(component.getPermissionsPath).toEqual([baseUrl, 'get']);     // TODO fails
+      expect(component.createPermissionsPath).toEqual([baseUrl, 'post']); // TODO fails
```

Comment:
I don't understand yet what I've done wrong here. The previous test setup seemed odd to me, I've tried making this more analogous to other specs. However, I've broken something in the process. Weirdly, too, the test below passes.

Reply:
`ngOnInit` is called when the component is initialized (in the `beforeEach` block), so setting `component.baseUrl` in your test comes too late. Several approaches to remedy.
If we expect `baseUrl` to change, then we need to accommodate it coming later--like in your test--which would be done with `ngOnChanges`. However, I think it is reasonable to say that we require `baseUrl` to be immutable--set only during instantation. (Probably should comment that!). 

More appropriate then, is to set `baseUrl` in your test _in the template_. Which, in a way, is more "real world" testing, too. Create a `TestHost`, complete with template.

A simple example: https://github.com/chef/automate/blob/0e6b4cf3b23f1ffb3bfaf3b5604ff1ddb9ca984b/components/automate-ui/src/app/components/message-modal/message-modal.component.spec.ts#L7-L22

A more elaborate example:
https://github.com/chef/automate/blob/e3ca5ae7a14539f2d097a542652fe0720cdeb9f4/components/automate-ui/src/app/components/form-control/form-control.directive.spec.ts#L63-L78

Reply:
Ok, so the odd thing I hadn't seen before (and thoughtlessly removed) was actually the right thing. Gotcha. Thanks.